ddtrace/contrib/internal/asgi/middleware.py

```diff
@@ -268,18 +224,20 @@ async def __call__(self, scope: Mapping[str, Any], receive: Callable, send: Call
             integration_config=config.asgi,
             activate_distributed_headers=True,
         ) as ctx, ctx.span as span:
-            span.set_tag_str(COMPONENT, self.integration_config.integration_name)
-            ctx.set_item("req_span", span)
+            # TODO: move into trace handlers
+            core.dispatch("asgi.__call__", (ctx, scope, self.integration_config))
```

Comment:
something I've tried for django, instead of doing `core.on("context.started.django.request", _start_span)` create a simple wrapper for it,

```python
def _on_django_request(ctx: core.ExecutionContext) -> None:
    span = _start_span(ctx)

    scope = ctx.get_local_item("scope")
    # ... do span stuff with the scope

core.on("context.started.django.request", _on_django_request)
```

then you can get rid of the double dispatch (`core.context_with_data` will call the `dispatch` for "context.started.{name}" and "context.ended.{name}")

Reply:
This dispatch and others are using `self.integration_config`, but the `core.context_with_data` does `integration_config=config.asgi`, is that correct?